framework/src/main/java/org/checkerframework/common/value/util/Range.java

```diff
@@ -41,96 +42,142 @@
+            if (min > current) min = current;
+            if (max < current) max = current;
+        }
+        return create(min, max);
     }
 
+    /** Long.MIN_VALUE, as a BigInteger. */
+    private static final BigInteger BIG_LONG_MIN_VALUE = BigInteger.valueOf(Long.MIN_VALUE);
+    /** Long.MAX_VALUE, as a BigInteger. */
+    private static final BigInteger BIG_LONG_MAX_VALUE = BigInteger.valueOf(Long.MAX_VALUE);
+    /** The number of Long values, as a BigInteger. */
+    private static final BigInteger BIG_LONG_WIDTH =
+            BIG_LONG_MAX_VALUE.subtract(BIG_LONG_MIN_VALUE).add(BigInteger.ONE);
+
     /**
-     * Returns a range with its bounds specified by two parameters, {@code from} and {@code to}. If
-     * {@code from} is greater than {@code to}, returns {@link #NOTHING}.
+     * Creates a range using BigInteger type bounds.
      *
-     * @param from the lower bound (inclusive)
-     * @param to the upper bound (inclusive)
+     * <p>If the BigInteger range is wider than the full range of the Long class, return EVERYTHING.
+     *
+     * <p>If one of the BigInteger bounds is out of Long's range and {@link #ignoreOverflow} is
+     * false, convert the bounds to Long type in accordance with Java twos-complement overflow
+     * rules, e.g., Long.MAX_VALUE + 1 is converted to Long.MIN_VALUE.
+     *
+     * <p>If one of the BigInteger bounds is out of Long's range and {@link #ignoreOverflow} is
+     * true, convert the bound that is outside Long's range to max/min value of a Long.
+     *
+     * @param bigFrom the lower bound of the BigInteger range
+     * @param bigTo the upper bound of the BigInteger range
+     * @return a range with Long type bounds converted from the BigInteger range
      */
-    private static Range createRangeOrNothing(long from, long to) {
-        if (from <= to) {
-            return new Range(from, to);
+    private static Range create(BigInteger bigFrom, BigInteger bigTo) {
+        if (ignoreOverflow) {
+            bigFrom = bigFrom.max(BIG_LONG_MIN_VALUE);
+            bigTo = bigTo.min(BIG_LONG_MAX_VALUE);
         } else {
-            return NOTHING;
+            BigInteger bigWidth = bigTo.subtract(bigFrom).add(BigInteger.ONE);
+            if (bigWidth.compareTo(BIG_LONG_WIDTH) > 0) {
+                return EVERYTHING;
+            }
         }
+        long longFrom = bigFrom.longValue();
+        long longTo = bigTo.longValue();
+        return createOrElse(longFrom, longTo, EVERYTHING);
     }
 
     /**
      * Returns a range with its bounds specified by two parameters, {@code from} and {@code to}. If
-     * {@code from} is greater than {@code to}, returns {@link #EVERYTHING}.
+     * {@code from} is greater than {@code to}, returns {@link #NOTHING}.
      *
      * @param from the lower bound (inclusive)
      * @param to the upper bound (inclusive)
-     * @return newly-created Range or EVERYTHING
+     * @return newly created Range or EVERYTHING
```

Comment:
Should the `EVERYTHING` be `NOTHING` as suggested by the method name?

Reply:
corrected